_docs/reference/config/kubernetes-proxy-injection.md

```diff
@@ -0,0 +1,214 @@
+    --injectConfigFile inject-config.yaml \
+    --meshConfigFile mesh-config.yaml \
+    --filename samples/sleep/sleep.yaml \
+    --output sleep-injected.yaml
+$ kubectl apply -f sleep-injected.yaml    
+```
+
+Alternatively, this can be performed in a single step. This uses the built-in default template and dynamically fetches the mesh configuration from the `istio` ConfigMap. Additional parameter overrides are available via flags (see `istioctl kube-inject --help`).
+
+```
+kubectl apply -f <(~istioctl kube-inject --meshConfigMapName=istio -f samples/sleep/sleep.yaml)
+```
+
+Verify that the sidecar has been injected into the deployment.
+
+```
+$ kubectl get deployment sleep -o wide
+NAME      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE       CONTAINERS          IMAGES                             SELECTOR
+sleep     1         1         1            1           2h        sleep,istio-proxy   tutum/curl,unknown/proxy:unknown   app=sleep
+```
+
+# Automatic injection with mutating webhooks
+
+See https://kubernetes.io/docs/admin/admission-controllers/#validatingadmissionwebhook-alpha-in-18-beta-in-19 and https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#mutatingwebhookconfiguration-v1beta1-admissionregistration for overview of webhook admission controller.
+
+## Prerequisites
+
+Kubernetes 1.9 cluster is required with `admissionregistration.k8s.io/v1beta1` enabled.
+
+```
+$ kubectl api-versions | grep admissionregistration.k8s.io/v1beta1
+admissionregistration.k8s.io/v1beta1
+```
+
+### GKE
+
+```bash
+gcloud container clusters create <cluster-name> \
+    --enable-kubernetes-alpha 
+    --cluster-version=1.9.1-gke.0 
+    --zone=<zone>
+    --project <project-name>
+    
+gcloud container clusters get-credentials <cluster-name> \
+    --zone <zone> \
+    --project <project-name>
+    
+kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole=cluster-admin \
+    --user=$(gcloud config get-value core/account)
+```
+
+### minikube
+### IBM Cloud Container Service
+### AWS with Kops
+
+## Installation
+
+
+Install base Istio.
+
+```
+$ kubectl apply -f install/kubernetes/istio.yaml
+```
+
+Webhooks requires a signed cert/key pair. Use `install/kubernetes/webhook-create-signed-cert.sh` to generate 
+a cert/key pair signed by the Kubernetes' CA. The resulting cert/key file is stored as a Kubernetes 
+secret for the sidecar injector webhook to consume.
+
+_Note_: Kubernetes CA approval requires permissions to create and approve CSR. See 
+https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster and 
+`install/kubernetes/webhook-create-signed-cert.sh` for more information.
+
+```
+$ ./install/kubernetes/webhook-create-signed-cert.sh \
+    --service istio-sidecar-injector \
+    --namespace istio-system \
+    --secret sidecar-injector-certs
+```
+
+Install the sidecar injection configmap. 
+
+```
+$ kubectl apply -f install/kubernetes/istio-sidecar-injector-configmap-debug.yaml
+```
```

Comment:
should we instruct to install release yaml here as that is the default with everything else?